test/app.test.ts

```diff
@@ -1,4 +1,360 @@
+          name: 'myname',
+        },
+      });
+
+      expect(response.statusCode).toEqual(StatusCodes.BAD_REQUEST);
+    });
+
+    it('throw if payload is empty', async () => {
+      const app = await build({
+        runner,
+        itemTaskManager,
+        memberTaskManager,
+        itemMembershipTaskManager,
+      });
+      const response = await app.inject({
+        method: 'PATCH',
+        url: `${item.id}/invitations/${invitation.id}`,
+        payload: {},
+      });
+
+      expect(response.statusCode).toEqual(StatusCodes.BAD_REQUEST);
+    });
+  });
+
+  describe('DELETE /:itemId/invitations/:id', () => {
+    const invitation = FIXTURES_INVITATIONS[0];
+    it('delete invitation successfully', async () => {
+      mockCreateDeleteInvitationTaskSequence(runner, invitation);
+
+      const app = await build({
+        runner,
+        itemTaskManager,
+        memberTaskManager,
+        itemMembershipTaskManager,
+      });
+      const response = await app.inject({
+        method: 'DELETE',
+        url: `${item.id}/invitations/${invitation.id}`,
+      });
+
+      expect(response.statusCode).toEqual(StatusCodes.OK);
+      expect(await response.json()).toEqual(invitation);
+    });
+
+    it('throw if item id is invalid', async () => {
+      const app = await build({
+        runner,
+        itemTaskManager,
+        memberTaskManager,
+        itemMembershipTaskManager,
+      });
+      const response = await app.inject({
+        method: 'DELETE',
+        url: `invalid/invitations/${invitation.id}`,
+      });
+
+      expect(response.statusCode).toEqual(StatusCodes.BAD_REQUEST);
+    });
+
+    it('throw if invitation id is invalid', async () => {
+      const app = await build({
+        runner,
+        itemTaskManager,
+        memberTaskManager,
+        itemMembershipTaskManager,
+      });
+      const response = await app.inject({
+        method: 'DELETE',
+        url: `${item.id}/invitations/invalid-id`,
+      });
+
+      expect(response.statusCode).toEqual(StatusCodes.BAD_REQUEST);
+    });
+  });
+
+  describe('POST /:itemId/invitations/:id/send', () => {
+    const invitation = FIXTURES_INVITATIONS[0];
+    it('send invitation successfully', async () => {
+      mockCreateDeleteInvitationTaskSequence(runner, invitation);
+
+      const buildInvitationLink = jest.fn().mockReturnValue('link');
+      const mockSendMail = jest.fn().mockImplementation(async () => {
+        const app = await build({
+          runner,
+          itemTaskManager,
+          memberTaskManager,
+          itemMembershipTaskManager,
+          sendInvitationEmail: mockSendMail,
+          buildInvitationLink,
+        });
+        const response = await app.inject({
+          method: 'POST',
+          url: `${item.id}/invitations/${invitation.id}/send`,
+        });
+
+        expect(response.statusCode).toEqual(StatusCodes.NO_CONTENT);
+        // check email got sent
+        setTimeout(() => {
```

Comment:
Info: why is the `setTimeout` needed?

Reply:
This function returns before sending the mails, so we wait a little bit for the mocks to execute.